### PR TITLE
CC-14561: Support multiple abstract products in discount promotional products.

### PR DESCRIPTION
**Pull request by romanhavrylko**
## PR Description

- Epic: https://spryker.atlassian.net/browse/CC-14561

- Ticket for IG: https://spryker.atlassian.net/browse/CC-14600

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+| --- | --- | --- |
+| Promotions & Discounts | {{page.version}} | [Promotions & Discounts feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/promotions-and-discounts-feature-integration.html) |
+| Spryker Cart | {{page.version}}   | [Spryker Cart feature integration](/docs/scos/dev/feature-integration-guides/{{page.version}}/cart-feature-integration.md.html) |
+
+### 1) Install the required modules using Composer
+
+Install the required modules:
+
+```bash
+composer require spryker/discount-promotions-rest-api "^1.4.0"
```
Comment (DmitryLymarenko) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:28]: that's not enough
at similar IG we updated feature

see https://github.com/spryker/spryker-docs/pull/684/files

please also discuss with Anton if you need to update basic IG as he did

Comment (DmitryLymarenko) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:28]: discussed

**Review by DmitryLymarenko — APPROVED** [review]

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Ensure that the plugin works correctly:
+
+1. [Create a discount](/docs/scos/user/back-office-user-guides/{{page.version}}/merchandising/discount/creating-cart-rules.html) and select **Promotional Product** radio button in **Discount Calculation** form.
+2. Add a product abstract sku into the **Abstract Product SKU(s)** field.
```
[the remark this answers is not in the document]
Comment (JacquieRooney) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:161]: If we're trying to validate that the plugin works, we should create a discount with multiple promo products, correct?

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+3. Create another discount with the at least one same **Promotional Product**.
```
Comment (JacquieRooney) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:162]: Is this required? Why are we asking them to create a discount with at least one of the same products as the other discount?

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+4. To fulfill the discounts' requirements, add items to the cart.
+5. Make sure that both discount are displayed in the the **Promotional Product** section on the cart page.
```
Comment (JacquieRooney) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:164]:
```suggestion
4. Make sure that all products from the discount are displayed in the the **Promotional Product** section on the cart page.
```

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+```
+
+{% info_block warningBox "Verification" %}
+
+Ensure that the plugin works correctly:
+
+1. [Create a discount](/docs/scos/user/back-office-user-guides/{{page.version}}/merchandising/discount/creating-cart-rules.html) and select **Promotional Product** radio button in **Discount Calculation** form.
+2. Add a product abstract sku into the **Abstract Product SKU(s)** field.
+3. Create another discount with the at least one same **Promotional Product**.
+4. To fulfill the discounts' requirements, add items to the cart.
```
Comment (JacquieRooney) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:163]:
```suggestion
3. Add items to the cart to fulfill the discount's conditions.
```

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 3 above)
Comment (JacquieRooney) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:162]:
```suggestion

```

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+Enable Javascript and CSS changes:
+
+```bash
+console frontend:zed:install-dependencies
+console frontend:zed:build
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that you can create a discount with multiple promotional products:
```
Comment (JacquieRooney) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:180]: Isn't this the same as the plugin verification process? Do we need this?

**Review by JacquieRooney — CHANGES_REQUESTED** [review]

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (romanhavrylko) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:161]: @JacquieRooney, a discount with only one promotion product is also fine. The main point here is to have two discounts with at least one same product. The plugin allows displaying the same product in several discount carousels on the cart page.

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 3 above)
Comment (romanhavrylko) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:162]: @JacquieRooney, the plugin allows displaying the same product in several discount carousels on the cart page.

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 6 above)
Comment (romanhavrylko) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:180]: @JacquieRooney, basically - yes. However, when we verify that the plugin works, our goal is to make sure that all promotional products are displayed on the cart page. While here we have to make sure that a discount is applied after a promotion product is added to the cart.
Should this step be added to the plugin verification instruction? So we can remove this section.

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JacquieRooney) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:161]: OK, got it. Please decline this change.

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 3 above)
Comment (JacquieRooney) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:162]: Understood, thanks for clarifying.

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md (hunk 6 above)
Comment (JacquieRooney) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:180]: No, we can leave it as-is. I understand the difference between the two exercises now.

**Review by JacquieRooney — DISMISSED** [review]

File: docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
+---
+title: Multiple Abstract Products as Promotional Products & Discounts feature integration
```
Comment (VadymSachenko) [inline docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md:2]: the title and file name are different. Which one is correct?
Both must be identic

Comment (romanhavrylko) [reply docs/scos/dev/feature-integration-guides/202200.0/multiple-abstract-products-promotions-and-discounts-feature-integration.md]: The title is correct

**Review by VadymSachenko — DISMISSED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Pull request merged**